### PR TITLE
Improve maxMinHelper in z/codegen 

### PR DESCRIPTION
**Pull request by dhong44**
This pull request improves the `maxMinHelper` evaluator
in Z codegen in 3 ways.
- Add guards for z10  in the evaluator and lower and disable max/min optimisations on unsupported platforms
- Rewrites the evaluator for lmin and lmax when using register pairs
- Add Tril tests to capture corner cases for this evaluator

### CODE REVIEW

**Comment by dhong44** [comment]
@fjeremic, @dchopra001

File: fvtest/compilertriltest/MaxMinTest.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int32_t (*)(void)>();
+    volatile auto exp = param.oracle(param.lhs, param.rhs);
+    volatile auto act = entry_point();
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/MaxMinTest.cpp]: Do these really need to be volatile? I think just having `ASSERT_EQ(eparam.oracle(param.lhs, param.rhs), entry_point())` should work.

File: fvtest/compilertriltest/MaxMinTest.cpp [hunk 2]
```diff
@@ -85,7 +120,7 @@ TEST_P(Int32MaxMin, UsingLoadParam) {
 
 
 INSTANTIATE_TEST_CASE_P(MaxMin, Int32MaxMin, ::testing::Combine(
-    ::testing::ValuesIn(TRTest::combine(TRTest::const_values<int32_t>(), TRTest::const_values<int32_t>())),
+    ::testing::ValuesIn(TRTest::combine(intMaxMinTestValues(), intMaxMinTestValues())),
```
Comment (fjeremic) [inline fvtest/compilertriltest/MaxMinTest.cpp]: Why not combine both of these? Or better yet why don't we scrap the two custom functions you have here (`intMaxMinTestValues` and `longMaxMinTestValues`) and put the constants directly in `TRTest::const_values` so all other tests get the benefit of testing the additional constants?

**Comment by dhong44** [comment]
These Trill tests will not work on z10 and lower since conditional loads are not supported. Currently Tril does not support platform specific testing, so we cannot guard against this. https://github.com/eclipse/omr/issues/1843 Is open to track this.

**Comment by fjeremic** [comment]
I'll preemptively launch the builds. However given the Tril changes I'll invite @Leonardo2718 to review again. Thanks for addressing all the concerns @dhong44!

@genie-omr build all

**Comment by fjeremic** [comment]
Travis AArch build failed due to infrastructure. Given this is a Z change I'm ok to merge.
